File: playwright/po/formValidationPO.js
const { expect } = require('@playwright/test');

exports.formValidationPO = class formValidationPO {
  /**
  * @param {import('@playwright/test').Page} page
  */
  constructor(page) {
  this.page = page;
  this.Title_text = page.locator("xpath=//h1[text()='Form Validation']");

  this.ContactName_input = page.locator("xpath=//input[@name='ContactName']");
  this.ContactNumber_input = page.locator("xpath=//input[@name='contactnumber']");
  this.PickUpDate_input = page.locator("xpath=//input[@name='pickupdate']");
  this.PaymentMethod_select = page.locator("xpath=//select[@name='payment']");
  this.Register_button = page.locator("xpath=//button[text()=' Register ']");
  //Errors
  this.ContactName_Empty_error = page.locator("xpath=//div[contains(text(),'Please enter your Contact name.')]");
  this.ContactName_LooksGood_message = page.locator("xpath=//div[contains(text(),'Looks good!')]");
  this.ContactNumber_Invalid_error = page.locator("xpath=//div[contains(text(),'Please provide your Contact number.')]");
  this.PickUpDate_Invalid_error = page.locator("xpath=//div[contains(text(),'Please provide valid Date.')]");
  this.PaymentMethod_Invalid_error = page.locator("xpath=//div[contains(text(),'Please select the Paymeny Method.')]");
  }
    
  async goto() {
    await this.page.goto('https://practice.expandtesting.com/form-validation');
    }

  async VerifyExpectedElementsAppearOnFormValidationPage() {
    await expect(this.Title_text).toBeVisible();
    await expect(this.ContactName_input).toBeVisible();
    await expect(this.ContactNumber_input).toBeVisible();
    await expect(this.PickUpDate_input).toBeVisible();
    await expect(this.PaymentMethod_select).toBeVisible();
    await expect(this.Register_button).toBeVisible();
    }

  //Contact Name
  async EnterContactName(value) {
    await this.ContactName_input.fill('');
    await this.ContactName_input.fill(value);
    }

  async ClearContactName() {
    await this.ContactName_input.fill('');
    }  

  async VerifyEmptyContactNameErrorAppears() {
    await expect(this.ContactName_Empty_error).toBeVisible();
    }

  async VerifyEmptyContactNameErrorDoesNotAppear() {
    await expect(this.ContactName_Empty_error).toBeHidden();
    }

  async VerifyLooksGoodContactNameMessageAppears() {
    await expect(this.ContactName_LooksGood_message).toBeVisible();
    }

  //Contact Number
  async EnterContactNumber(value) {
    await this.ContactNumber_input.fill('');
    await this.ContactNumber_input.fill(value);
  }

  async ClearContactNumber() {
    await this.ContactNumber_input.fill('');   
  }      

  async VerifyInvalidContactNumberErrorAppears() {
    await expect(this.ContactNumber_Invalid_error).toBeVisible();
  }  

  async VerifyInvalidContactNumberErrorDoesNotAppear() {
    await expect(this.ContactNumber_Invalid_error).toBeHidden();
  }  

  //PickUp Date
  async EnterPickUpDate(value) {
    await this.PickUpDate_input.fill('');
    await this.PickUpDate_input.fill(value);
  }      

  async ClearPickUpDate() {
    await this.PickUpDate_input.fill(''); 
  }    

  async VerifyInvalidPickUpDateErrorAppears() {
    await expect(this.PickUpDate_Invalid_error).toBeVisible();
  }    

  async VerifyInvalidPickUpDateErrorDoesNotAppear() {
    await expect(this.PickUpDate_Invalid_error).toBeHidden();
  }    

  //Payment Method
  async SelectPaymentMethod(value) {
    await this.PaymentMethod_select.selectOption(value);
  }    

  async VerifyInvalidPaymentMethodErrorAppears() {
    await expect(this.PaymentMethod_Invalid_error).toBeVisible();
  }    

  async VerifyInvalidPaymentMethodErrorDoesNotAppear() {
    await expect(this.PaymentMethod_Invalid_error).toBeHidden();
  }    

  //Register
  async ClickRegister() {
    await this.Register_button.click(); 
  }

  };

File: java-selenium/javaSelenium/src/main/java/pageObjects/formValidationPO.java
package pageObjects;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.Select;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class formValidationPO {

	WebDriver driver;	
	By FormValidationPage_Title_text=By.xpath("//h1[text()='Form Validation']");
	By FormValidationPage_ContactName_input=By.xpath("//input[@name='ContactName']");
	By FormValidationPage_ContactNumber_input=By.xpath("//input[@name='contactnumber']");
	By FormValidationPage_PickUpDate_input=By.xpath("//input[@name='pickupdate']");
	By FormValidationPage_PaymentMethod_select=By.xpath("//select[@name='payment']");
	By FormValidationPage_Register_button=By.xpath("//button[text()=' Register ']");
	//Errors
	By FormValidationPage_ContactName_Empty_error=By.xpath("//div[contains(text(),'Please enter your Contact name.')]");
	By FormValidationPage_ContactName_LooksGood_message=By.xpath("//div[contains(text(),'Looks good!')]");
	By FormValidationPage_ContactNumber_Invalid_error=By.xpath("//div[contains(text(),'Please provide your Contact number.')]");
	By FormValidationPage_PickUpDate_Invalid_error=By.xpath("//div[contains(text(),'Please provide valid Date.')]");
	By FormValidationPage_PaymentMethod_Invalid_error=By.xpath("//div[contains(text(),'Please select the Paymeny Method.')]");
	
	public formValidationPO(WebDriver driver) {
		this.driver=driver;
	}
	
	public void VerifyExpectedElementsAppearOnFormValidationPage() {
		assertTrue(driver.findElement(FormValidationPage_Title_text).isDisplayed());
		assertTrue(driver.findElement(FormValidationPage_ContactName_input).isDisplayed());
		assertTrue(driver.findElement(FormValidationPage_ContactNumber_input).isDisplayed());
		assertTrue(driver.findElement(FormValidationPage_PickUpDate_input).isDisplayed());
		assertTrue(driver.findElement(FormValidationPage_PaymentMethod_select).isDisplayed());
		assertTrue(driver.findElement(FormValidationPage_Register_button).isDisplayed());
	}

	//Contact Name
	public void EnterContactName(String value){
		driver.findElement(FormValidationPage_ContactName_input).clear();
		driver.findElement(FormValidationPage_ContactName_input).sendKeys(value);
	}
	
	public void ClearContactName() {
		driver.findElement(FormValidationPage_ContactName_input).clear();
	} 

	public void VerifyEmptyContactNameErrorAppears() {
		assertTrue(driver.findElement(FormValidationPage_ContactName_Empty_error).isDisplayed());
	}

	public void VerifyLooksGoodContactNameMessageAppears() {
		assertTrue(driver.findElement(FormValidationPage_ContactName_LooksGood_message).isDisplayed());
	}
	
	//Contact Number
	public void EnterContactNumber(String value) {
		driver.findElement(FormValidationPage_ContactNumber_input).clear();
		driver.findElement(FormValidationPage_ContactNumber_input).sendKeys(value);
	}

    public void ClearContactNumber () {   
		driver.findElement(FormValidationPage_ContactNumber_input).clear();
	}

    public void VerifyInvalidContactNumberErrorAppears() {
		assertTrue(driver.findElement(FormValidationPage_ContactNumber_Invalid_error).isDisplayed());
	}

    public void VerifyInvalidContactNumberErrorDoesNotAppear() {
		assertTrue(!driver.findElement(FormValidationPage_ContactNumber_Invalid_error).isDisplayed());
	}
    
	//PickUp Date
    public void EnterPickUpDate(String value) {
		driver.findElement(FormValidationPage_PickUpDate_input).clear();
		driver.findElement(FormValidationPage_PickUpDate_input).sendKeys(value);
    }
    
    public void ClearPickUpDate() {
		driver.findElement(FormValidationPage_PickUpDate_input).clear();
    }
    
    public void VerifyInvalidPickUpDateErrorAppears() {
		assertTrue(driver.findElement(FormValidationPage_PickUpDate_Invalid_error).isDisplayed());
	}
    
    public void VerifyInvalidPickUpDateErrorDoesNotAppear() {
		assertTrue(!driver.findElement(FormValidationPage_PickUpDate_Invalid_error).isDisplayed());
	}
    
	//Payment Method
	
	//Register
	public void ClickRegister() {
		driver.findElement(FormValidationPage_Register_button).click();
	}

	public void SelectPaymentMethod(String value) { 
		WebElement dropdownElement = driver.findElement(FormValidationPage_PaymentMethod_select);
		Select dropdownSelect = new Select(dropdownElement);
		dropdownSelect.selectByVisibleText(value);
	}

	public void VerifyInvalidPaymentMethodErrorDoesNotAppear() {
		assertTrue(!driver.findElement(FormValidationPage_PaymentMethod_Invalid_error).isDisplayed());
		
	}

	public void VerifyInvalidPaymentMethodErrorAppears() {
		assertTrue(driver.findElement(FormValidationPage_PaymentMethod_Invalid_error).isDisplayed());
		
	}
}
